### PR TITLE
GG-40379 CPP: Add BinaryReader::IsNull and BinaryRawReader::SkipIfNull

### PR DESCRIPTION
**Pull request by isapego**
https://ggsystems.atlassian.net/browse/GG-40379

- Added `BinaryReader::IsNull` and `BinaryRawReader::SkipIfNull` methods to check for `null`s;
- Modified tests to check new methods.

### CODE REVIEW

File: modules/platforms/cpp/binary/include/ignite/binary/binary_raw_reader.h [hunk 1]
```diff
@@ -64,7 +64,14 @@ namespace ignite
              * @param impl Implementation.
              */
             BinaryRawReader(ignite::impl::binary::BinaryReaderImpl* impl);
-                        
+
+            /**
+             * Skip next value if it is the null.
```
Comment (ptupitsyn) [inline modules/platforms/cpp/binary/include/ignite/binary/binary_raw_reader.h:69]:
```suggestion
             * Skip the next value if it is null.
```

File: modules/platforms/cpp/binary/include/ignite/binary/binary_raw_reader.h [hunk 2]
```diff
@@ -64,7 +64,14 @@ namespace ignite
...
+             *
+             * @return True if the null value has been detected and skipped.
```
Comment (ptupitsyn) [inline modules/platforms/cpp/binary/include/ignite/binary/binary_raw_reader.h:71]:
```suggestion
             * @return True if a null value has been detected and skipped.
```

File: modules/platforms/cpp/binary/src/impl/binary/binary_reader_impl.cpp [hunk 3]
```diff
@@ -777,6 +777,24 @@ namespace ignite
...
+
+                int32_t fieldId = idRslvr->GetFieldId(typeId, fieldName);
+                int32_t fieldPos = FindField(fieldId);
+
+                if (fieldPos <= 0)
+                    return true;
+
+                stream->Position(fieldPos);
+
+                return SkipIfNull();
```
Comment (ptupitsyn) [inline modules/platforms/cpp/binary/src/impl/binary/binary_reader_impl.cpp:795]: `SkipIfNull` will advance the position, is that ok here?

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/platforms/cpp/binary/src/impl/binary/binary_reader_impl.cpp (hunk 3 above)
Comment (isapego) [reply modules/platforms/cpp/binary/src/impl/binary/binary_reader_impl.cpp:795]: Yeah, there is a position guard in the beginning. It makes sure the position is going to be restored when the function is left.

File: modules/platforms/cpp/binary/include/ignite/binary/binary_raw_reader.h (hunk 1 above)
Comment (isapego) [reply modules/platforms/cpp/binary/include/ignite/binary/binary_raw_reader.h:69]: Yeah, copied old comment without checking. My bad. Fixed.

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**